### ISSUE TITLE
Add specific error message during "card declined" errors when using the blocks-based checkout

### ISSUE THREAD

**Issue by reykjalin** [Automattic/woocommerce-payments#1706]
## What I expected

When my card is declined I expect that I'll get an error message telling me the card was declined.

## What happened instead

> When I tried to pay **through the credit card form** with a card that will be declined (`4000 0000 0000 0002`) I don't get the expected `Card declined` error, and instead get this generic message at the top of the checkout page:
>
> | Error Message | Credit Card Form |
> | :---: | :---: |
> | <img width="832" alt="image" src="https://user-images.githubusercontent.com/13835680/116463974-e6584680-a85a-11eb-85a8-9f9431803fea.png"> | <img width="528" alt="image" src="https://user-images.githubusercontent.com/13835680/116464001-f1ab7200-a85a-11eb-82f3-157f030a1d2c.png"> |
>
> Originally posted by @reykjalin in https://github.com/Automattic/woocommerce-payments/pull/1529#pullrequestreview-647456481

I'm not sure if this applies to more card issues, but it'd be worth it to check the error message from all the different types of error responses we handle.

## Reproduction Steps

1. Make sure you're using the Blocks checkout.
2. Add 1 or more products to your cart.
3. Go to the blocks checkout page
4. Pay with a card that will be declined, e.g. `4000 0000 0000 0002`.
5. Observe the generic error message.

**Comment by a-danae** [comment]
This doesn't seem to be an issue from WCPay but from WooCommerce Blocks.

We're adding the process' error response as a notice [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L804). Then on WooCommerce Blocks, we clear the notices without displaying them [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/Payments/Api.php#L169). 

There's a comment right above that line saying:
> If `process_payment` added notices, clear them. Notices are not displayed from the API -- payment should fail, and a generic notice will be shown instead if payment failed.

So, looks like it's intentional? I agree with this issue, it'd be more useful to display the actual errors from `process_payment` instead of a generic message. But I'm not aware of the reasoning behind the current approach.

This seems to be solved by adding `NoticeHandler::convert_notices_to_exceptions( 'woocommerce_rest_payment_error' );` after the `process_payment()` call in WooCommerce Blocks, [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/Payments/Api.php#L169), replacing `wc_clear_notices()`.

@reykjalin what do you think? Not sure how to proceed from here. If that makes sense, I guess we could move this issue to WooCommerce Blocks' repo, and I can create a PR there.

**Comment by a-danae** [comment]
Copied this issue to the WooCommerce Block repo. Closing this issue in favor of [that other one](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4870).
